packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -0,0 +1,634 @@
+      const apiKeyDescription = apiKeyConfig!.apiKeyConfig?.description;
+      const apiKeyExpirationDays = apiKeyConfig!.apiKeyConfig?.expires;
+
+      const apiKey = api.createAPIKey({
+        description: apiKeyDescription,
+        expires: apiKeyExpirationDays,
+      });
+
+      new CfnOutput(rootStack, 'GraphQLAPIKeyOutput', {
+        value: apiKey.attrApiKey,
+        description: 'Your GraphQL API ID.',
+        exportName: Fn.join(':', [Aws.STACK_NAME, 'GraphQLApiKey']),
+      });
+    }
+
+    new CfnOutput(rootStack, 'GraphQLAPIIdOutput', {
+      value: api.apiId,
+      description: 'Your GraphQL API ID.',
+      exportName: Fn.join(':', [Aws.STACK_NAME, 'GraphQLApiId']),
+    });
+
+    new CfnOutput(rootStack, 'GraphQLAPIEndpointOutput', {
+      value: api.graphqlUrl,
+      description: 'Your GraphQL API endpoint.',
+      exportName: Fn.join(':', [Aws.STACK_NAME, 'GraphQLApiEndpoint']),
+    });
+    api.addToSchema(output.buildSchema());
+    return api;
+  }
+
+  private synthesize(context: TransformerContext): DeploymentResources {
+    const stackManager: StackManager = context.stackManager as StackManager;
+    // eslint-disable-next-line no-unused-expressions
+    this.app?.synth({ force: true, skipValidation: true });
+
+    const templates = stackManager.getCloudFormationTemplates();
+    const rootStackTemplate = templates.get('transformer-root-stack');
+    let childStacks: Record<string, Template> = {};
+    for (let [templateName, template] of templates.entries()) {
+      if (templateName !== 'transformer-root-stack') {
+        childStacks[templateName] = template;
+      }
+    }
+
+    const fileAssets = stackManager.getMappingTemplates();
+    const pipelineFunctions: Record<string, string> = {};
+    const resolvers: Record<string, string> = {};
+
+    for (let [templateName, template] of fileAssets) {
+      if (templateName.startsWith('pipelineFunctions/')) {
+        pipelineFunctions[templateName.replace('pipelineFunctions/', '')] = template;
+      } else if (templateName.startsWith('resolvers/')) {
+        resolvers[templateName.replace('resolvers/', '')] = template;
+      }
+    }
+    const schema = fileAssets.get('schema.graphql') || '';
+    return {
+      functions: {},
+      pipelineFunctions,
+      stackMapping: {},
+      resolvers,
+      schema,
+      stacks: childStacks,
+      rootStack: rootStackTemplate!,
+    };
+  }
+
+  private collectResolvers(context: TransformerContext, api: GraphQLAPIProvider): void {
+    const resolverEntries = context.resolvers.collectResolvers();
+    for (let [, resolver] of resolverEntries) {
+      resolver.synthesize(context, api);
+    }
+  }
+
+  private transformObject(
+    transformer: TransformerPluginProvider,
+    def: ObjectTypeDefinitionNode,
+    validDirectiveNameMap: { [k: string]: boolean },
+    context: TransformerContext,
+  ) {
+    let index = 0;
+    for (const dir of def.directives ?? []) {
```

Comment:
instead of maintaining index separately, you can use:
```suggestion
    for (const [index, dir] of (def.directives ?? []).entries()) {
```